packages/block-editor/src/store/reducer.js

```diff
@@ -1043,6 +1043,25 @@ export function blockSelection( state = BLOCK_SELECTION_INITIAL_STATE, action )
 	return state;
 }
 
+/**
+ * Reducer returning the block zoom's state.
+ *
+ * @param {Object} state  Current state.
+ * @param {Object} action Dispatched action.
+ *
+ * @return {Object} Updated state.
+ */
+export function blockZoom( state, action ) {
+	switch ( action.type ) {
+		case 'ZOOM_BLOCKS':
+			return action.clientIds;
+		case 'CLEAR_ZOOMED_BLOCKS':
```

Comment:
Should we add cases for remove and replace block actions?

Reply:
Yes.